### PR TITLE
Connect Api to Project

### PR DESCRIPTION
**Pull request by jodavid444**
# In This Pull Request:

- Make use of Axios to communicate with the backend and to make API calls to the project

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,27 +1,43 @@
...
+  } catch (error) {
+    throw new Error(error);
   }
-};
+});
 
-export const addBook = (book) => ({
-  type: ADD_BOOK,
-  book,
+export const addBook = createAsyncThunk(ADD_BOOK, async (payload, thunkAPI) => {
```
Comment (topeogunleye) [inline src/redux/books/books.js:17]: - [ ] I kindly suggest that `addBook` and `removeBook` should still work after implementing your API. Kindly see the video below for more details:

https://www.loom.com/share/556223a85ead40bb85b1c874c34c7b52

File: src/redux/books/books.js [hunk 2]
```diff
@@ -1,27 +1,43 @@
...
+  try {
+    await BookApi.postBookApi(payload);
+    thunkAPI.dispatch(loadBooks());
+  } catch (error) {
+    throw new Error(error);
+  }
+});
+
+export const removeBook = createAsyncThunk(REMOVE_BOOK, async (payload, thunkAPI) => {
```
Comment (topeogunleye) [inline src/redux/books/books.js:26]: - [ ] I kindly suggest that `addBook` and `removeBook` should still work after implementing your API. Kindly see the video below for more details:

https://www.loom.com/share/556223a85ead40bb85b1c874c34c7b52

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @jodavid444, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

> ## (Highlights) Good Points: 💪 
> * Good commit messages. :heavy_check_mark:
> * No linter errors. :heavy_check_mark:
> * Followed Github flow :heavy_check_mark:
> * Professional README file. :heavy_check_mark:
> * PR has a good title :heavy_check_mark:

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Nothing to mention

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 2 above)
Comment (jodavid444) [reply src/redux/books/books.js:26]: I have fix the problem the application is working properly now

**Review by victorokoroji — APPROVED** [review]
Hi @jodavid444, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request is following the Git Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The commit messages are meaningful ✅ 
- Add and remove function working properly ✅ 
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by jodavid444** [comment]
Thank you for the approval and keep up the good work.  👍  @vickymarz

File: src/redux/books/books.js (hunk 2 above)
Comment (topeogunleye) [reply src/redux/books/books.js:26]: > I have fix the problem the application is working properly now

Yes

**Pull request merged**